localstack/utils/aws/aws_stack.py

```diff
@@ -327,8 +327,13 @@ def fix_account_id_in_arns(response, colon_delimiter=':', existing=None, replace
 
 
 def get_s3_client():
+    if 'S3_BACKEND' in os.environ:
+        endpoint_url = os.environ.get('S3_BACKEND')
```

Comment:
In certain cases, the environment variable `S3_BACKEND` could be set to an empty string (which would result in an error here). To make the code a bit more resilient, can we please change this to:
```
def get_s3_client():
    endpoint_url = os.environ.get('S3_BACKEND', '').strip() or config.TEST_S3_URL
    ...
```